test/e2e/page-objects/pages/account-list-page.ts

```diff
@@ -0,0 +1,92 @@
+  private driver: Driver;
+
+  private accountListItem: object;
+
+  private accountOptionsMenuButton: string;
+
+  private hideUnhideAccountButton: string;
+
+  private pinUnpinAccountButton: string;
+
+  private hiddenAccountsList: string;
+
+  private pinnedIcon: string;
+
+  private hiddenAccountOptionsMenuButton: string;
+
+  constructor(driver: Driver) {
+    this.driver = driver;
+    this.accountOptionsMenuButton =
+      '[data-testid="account-list-item-menu-button"]';
+    this.hideUnhideAccountButton = '[data-testid="account-list-menu-hide"]';
+    this.pinUnpinAccountButton = '[data-testid="account-list-menu-pin"]';
+    this.hiddenAccountsList = '[data-testid="hidden-accounts-list"]';
+    this.pinnedIcon = '[data-testid="account-pinned-icon"]';
+    this.accountListItem = {
+      text: 'Account',
+      css: '.multichain-account-menu-popover__list--menu-item',
+    };
+    this.hiddenAccountOptionsMenuButton =
+      '.multichain-account-menu-popover__list--menu-item-hidden-account [data-testid="account-list-item-menu-button"]';
+  }
+
+  async openAccountOptionsMenu(): Promise<void> {
+    console.log(`Open account option menu`);
+    await this.driver.clickElement(this.accountOptionsMenuButton);
+  }
+
+  async hideAccount(): Promise<void> {
+    console.log(`Hide account in account list`);
+    await this.driver.clickElement(this.hideUnhideAccountButton);
+  }
+
+  async check_hiddenAccountsListExists(): Promise<void> {
+    console.log(`Check that hidden accounts list is displayed in account list`);
+    await this.driver.waitForSelector(this.hiddenAccountsList);
+  }
+
+  async openHiddenAccountsList(): Promise<void> {
+    console.log(`Open hidden accounts option menu`);
+    await this.driver.clickElement(this.hiddenAccountsList);
+  }
+
+  async openHiddenAccountOptions(): Promise<void> {
+    console.log(`Open hidden accounts options menu`);
+    await this.driver.clickElement(this.hiddenAccountOptionsMenuButton);
+  }
+
+  async unhideAccount(): Promise<void> {
+    console.log(`Unhide account in account list`);
+    await this.driver.clickElement(this.hideUnhideAccountButton);
+  }
+
+  async check_accountIsDisplayed(): Promise<void> {
+    console.log(`Check that account is displayed in account list`);
+    await this.driver.waitForSelector(this.accountListItem);
+  }
+
+  async pinAccount(): Promise<void> {
+    console.log(`Pin account in account list`);
+    await this.driver.clickElement(this.pinUnpinAccountButton);
+  }
+
+  async unpinAccount(): Promise<void> {
+    console.log(`Unpin account in account list`);
+    await this.driver.clickElement(this.pinUnpinAccountButton);
+  }
+
+  async check_accountIsPinned(): Promise<void> {
+    console.log(`Check that account is pinned`);
+    await this.driver.waitForSelector(this.pinnedIcon);
+  }
+
+  async check_accountIsUnpinned(): Promise<void> {
+    console.log(`Check that account is unpinned`);
+    await this.driver.assertElementNotPresent(this.pinnedIcon);
+  }
```

Comment:
nit: for readibility, maybe we can order the methods alphabetically? I think it's okay to leave all the `check_XYZ` at the bottom though

Reply:
Good suggestion! I asked Devin to order them in this commit: [bf3f406](https://github.com/MetaMask/metamask-extension/pull/27201/commits/bf3f406edfb203b3566b2bb98c948b3b7faeaeee)